enh: add a list of known nvidia drivers as an attr in nixGL

cfhammill opened the pull request:
This can be used to generate wrappers purely and programmatically inside of flakes.

There's a script I used to generate the version/hash pairs in the list, see the included readme for more details.

cfhammill committed:
enh: add a list of known nvidia drivers as an attr in nixGL

This can be used to generate wrappers purely and programmatically
inside of flakes

joefiorini commented:
I like this! Just had to spend 2 hours or so updating a fork of this code because my distro updated my nvidia driver to 530. Although it now needs `zstd`, so I had to modify the dependencies of the fetcher as well.

cfhammill committed:
enh: update driver list, update script to incrementally add

SuperSandro2000 commented on known_drivers/mk-nvidia-drivers.py:
```diff
@@ -0,0 +1,69 @@
+                print("Download Failed")
+                pass
+    print("]", file=f)
+
+
+
+
+
```
```suggestion
```

SuperSandro2000 commented on known_drivers/mk-nvidia-drivers.py:
```diff
@@ -0,0 +1,69 @@
+                if CLEANUP:
+                    print(f"Attempting cleanup of {store_pth}")
+                    try:
+                        subprocess.check_output(["nix-store", "--delete", store_pth])
+                        print("Cleanup success.")
```
Why not just run nix-collect-garbage?

cfhammill replied:
in my own nix use I'm not very careful with my gcroots and would prefer the gc not be run unless I do it explicitly. Do you see advantages to using that instead?

SuperSandro2000 replied:
It just feels unnecessary if there are easier ways.

cfhammill replied:
do you mean running nix-collect-garbage after downloading each runfile? If so I don't see if as much of an improvement and may inadvertently gc something unintended. 

If you mean wait until the end of the script and then call gc, if so that's probably a bit faster, the store deletion adds some overhead to the download loop, but still runs the risk of gc'ing something unintended. Further the runfiles are now quite large and on computers with limited space waiting to the end to gc might not be best, this was why I wrote this in in the first place.

If you mean leave it up to the user to GC or not after running this script, that sounds good, suffers the space issue noted above, but in most contexts, especially with incremental downloading is probably fine.

SuperSandro2000 replied:
> do you mean running nix-collect-garbage after downloading each runfile? If so I don't see if as much of an improvement and may inadvertently gc something unintended.

Just once in the end if desired and I would be willing to accept that.

bcdarwin replied:
I'm happy with this approach; it seems more fine-grained and more performant than calling nix-collect-garbage, and you could always disabled cleanup and call the gc yourself if preferred.

SuperSandro2000 commented on known_drivers/mk-nvidia-drivers.py:
```diff
@@ -0,0 +1,69 @@
+
+if os.path.exists("driver-versions.nix"):
+    print("driver-versions.nix already exists, parsing known versions to skip re-downloads.")
+    print("copying old versions to driver-versions.bak.nix.")
+    shutil.copyfile("driver-versions.nix", "driver-versions.bak.nix")
```
We are using git, so this is not required

cfhammill replied:
good point, will remove

cfhammill committed:
cleanup: remove unnecessary filecopy and trailing whitespace

cfhammill committed:
enh: add 535.86.05

Commit:
add driver versions up to 20230927

Commit:
add driver versions 20231109

Commit:
convert mk-nvidia-drivers to a nix-shell script

Commit:
add driver versions 20240124

cfhammill commented:
Now that this project has moved to nix-community, is anyone with commit rights able to say if they'd be interested in ever merging this? 

I'm maintaining the branch for my own projects, no need to leave the PR open if this addition isn't desirable for the nixGL userbase at large.

ulucs commented:
Hello! Is there any way for me to help let this PR get merged?

Commit:
add driver versions 2024-12-06

Commit:
convert to dumping drivers to json

refactor a bit

Commit:
read drivers from json, implement known url mechanism

removes the nix file with driver versions

Commit:
add datacentre drivers, support fetching from nvidia

Commit:
add the dynamic nixGL wrapper

bcdarwin commented:
> Overall this looks great! We can follow up later by adding these to the tests. I had a couple of minor questions but I think this could be merged as-is or after one run to update the current list of drivers.

Any objection to merging as-is and updating the driver set in a new PR?

Commit:
known drivers: add timeout in prefetch command

ran into an issue where a single download hung
indefinitely. Cancelled the script and had to start
over from scratch. It is preferrable to skip that version
to complete writing out to driver-versions.json so that
other downloads can be skipped moving forward.

Commit:
known drivers: 2026-01-23 update

Commit:
add driver versions 2026-06-01

bcdarwin committed:
driver-versions.json: fix JSON syntax error

cfhammill committed:
Merge pull request #2 from bcdarwin/fix-syntax-error

driver-versions.json: fix JSON syntax error

Commit:
fix: retrieve driver version from driver file

chloepouprom committed:
Update dynamicWrapper.nix

Co-authored-by: Chris Hammill <cfhammill@gmail.com>

chloepouprom committed:
Update dynamicWrapper.nix

Co-authored-by: Chris Hammill <cfhammill@gmail.com>

Commit:
remove gnused

cfhammill committed:
Merge pull request #3 from chloepouprom/main

Fix nvidia module file parsing
